core/src/main/java/io/questdb/cairo/wal/ApplyWal2TableJob.java

```diff
@@ -486,91 +493,83 @@ private void storeWalTelemetryNoop(short event, TableToken tableToken, int walId
+                    lastWriterTxn = writer.getSeqTxn();
+                } catch (EntryUnavailableException tableBusy) {
+                    //noinspection StringEquality
+                    if (tableBusy.getReason() != NO_LOCK_REASON
+                            && !WAL_2_TABLE_WRITE_REASON.equals(tableBusy.getReason())
+                            && !WAL_2_TABLE_RESUME_REASON.equals(tableBusy.getReason())) {
+                        LOG.critical().$("unsolicited table lock [table=").utf8(tableToken.getDirName()).$(", lockReason=").$(tableBusy.getReason()).I$();
+                        // This is abnormal termination but table is not set to suspended state.
+                        // Reset state of SeqTxnTracker so that next CheckWalTransactionJob run will send job notification if necessary.
+                        engine.notifyWalTxnRepublisher(tableToken);
+                    }
+                    // Do not suspend table. Perhaps writer will be unlocked with no transaction applied.
+                    // We do not suspend table because of having initial value on lastWriterTxn. It will either be
+                    // "ignore" or last txn we applied.
+                    return;
                 }
-                // Don't suspend table. Perhaps writer will be unlocked with no transaction applied.
-                // We don't suspend table by virtue of having initial value on lastWriterTxn. It will either be
-                // "ignore" or last txn we applied.
-                return lastWriterTxn;
-            }
 
-            if (engine.getTableSequencerAPI().notifyCommitReadable(tableToken, lastWriterTxn)) {
-                engine.notifyWalTxnCommitted(tableToken);
+                if (engine.getTableSequencerAPI().notifyCommitReadable(tableToken, lastWriterTxn)) {
+                    engine.notifyWalTxnCommitted(tableToken);
+                }
             }
         } catch (CairoException ex) {
             if (ex.isTableDropped() || engine.isTableDropped(tableToken)) {
                 engine.notifyDropped(tableToken);
                 // Table is dropped, and we received cairo exception in the middle of apply
-                return tryDestroyDroppedTable(tableToken, null, engine, tempPath) ? Long.MAX_VALUE : WAL_APPLY_IGNORE_ERROR;
+                tryDestroyDroppedTable(tableToken, null, engine, tempPath);
+            } else {
+                telemetryFacade.store(TelemetrySystemEvent.WAL_APPLY_SUSPEND, TelemetryOrigin.WAL_APPLY);
+                LOG.critical().$("job failed, table suspended [table=").utf8(tableToken.getDirName())
+                        .$(", error=").$(ex.getFlyweightMessage())
+                        .$(", errno=").$(ex.getErrno())
+                        .I$();
+                handleWalApplyFailure(tableToken, ex.isOutOfMemory()
+                        ? new WalError(ex.getErrno(), Tag.FAILED_MEMORY_ALLOCATION, ex.getFlyweightMessage())
```

Comment:
this is a bug, we are holding on to mutable CharSequence that is liable to change spuriously.

Reply:
fixed